src/wp-includes/post.php

```diff
@@ -1440,31 +1444,36 @@ function _post_type_meta_capabilities( $capabilities = null ) {
+		'edit_item' => array( __('Edit ' . $singular_name), __('Edit Page') ),
+		'new_item' => array( __('New ' . $singular_name), __('New Page') ),
+		'view_item' => array( __('View ' . $singular_name), __('View Page') ),
+		'view_items' => array( __('View ' . $plural_name), __('View Pages') ),
+		'search_items' => array( __('Search ' . $plural_name), __('Search Pages') ),
+		'not_found' => array( __('No ' . $plural_name . 'found.'), __('No pages found.') ),
+		'not_found_in_trash' => array( __('No ' . $plural_name . 'found in Trash.'), __('No pages found in Trash.') ),
 		'parent_item_colon' => array( null, __('Parent Page:') ),
-		'all_items' => array( __( 'All Posts' ), __( 'All Pages' ) ),
-		'archives' => array( __( 'Post Archives' ), __( 'Page Archives' ) ),
-		'attributes' => array( __( 'Post Attributes' ), __( 'Page Attributes' ) ),
-		'insert_into_item' => array( __( 'Insert into post' ), __( 'Insert into page' ) ),
-		'uploaded_to_this_item' => array( __( 'Uploaded to this post' ), __( 'Uploaded to this page' ) ),
-		'featured_image' => array( _x( 'Featured Image', 'post' ), _x( 'Featured Image', 'page' ) ),
-		'set_featured_image' => array( _x( 'Set featured image', 'post' ), _x( 'Set featured image', 'page' ) ),
-		'remove_featured_image' => array( _x( 'Remove featured image', 'post' ), _x( 'Remove featured image', 'page' ) ),
-		'use_featured_image' => array( _x( 'Use as featured image', 'post' ), _x( 'Use as featured image', 'page' ) ),
-		'filter_items_list' => array( __( 'Filter posts list' ), __( 'Filter pages list' ) ),
-		'items_list_navigation' => array( __( 'Posts list navigation' ), __( 'Pages list navigation' ) ),
-		'items_list' => array( __( 'Posts list' ), __( 'Pages list' ) ),
+		'all_items' => array( __( 'All ' . $plural_name ), __( 'All Pages' ) ),
+		'archives' => array( __( $singular_name . ' Archives' ), __( 'Page Archives' ) ),
+		'attributes' => array( __( $singular_name . ' Attributes' ), __( 'Page Attributes' ) ),
+		'insert_into_item' => array( __( 'Insert into ' . $singular_name ), __( 'Insert into page' ) ),
+		'uploaded_to_this_item' => array( __( 'Uploaded to this ' . $singular_name ), __( 'Uploaded to this page' ) ),
+		'featured_image' => array( _x( 'Featured Image', $singular_name ), _x( 'Featured Image', 'page' ) ),
+		'set_featured_image' => array( _x( 'Set featured image', $singular_name ), _x( 'Set featured image', 'page' ) ),
+		'remove_featured_image' => array( _x( 'Remove featured image', $singular_name ), _x( 'Remove featured image', 'page' ) ),
+		'use_featured_image' => array( _x( 'Use as featured image', $singular_name ), _x( 'Use as featured image', 'page' ) ),
+		'filter_items_list' => array( __( 'Filter ' . $plural_name . ' list' ), __( 'Filter pages list' ) ),
+		'items_list_navigation' => array( __( $plural_name . ' list navigation' ), __( 'Pages list navigation' ) ),
+		'items_list' => array( __( $plural_name . ' list' ), __( 'Pages list' ) ),
```

Comment:
Actually, now that I look at it again, this specific approach is going to be a problem for translations.  Translations are parsed statically (they have to be otherwise it's impossible to know what's going to end up inside the `__( )` calls).

Reply:
Thanks for the review. Let me look into the static parsing. Maybe @Mte90 could advise a way around. The reason why I would go for a filter, is that when one adds it, they are doing it in their languages. I also presumed that the initial words stay as Post and Posts that can be translated.

Reply:
Well I already do this without any filter.
In https://codex.wordpress.org/Function_Reference/register_post_type#Parameters check the label/labels parameter that already support `singular_name` and `name` (that is plural).
Also this can break retrocompatibility also with other plugin and is not consistent with the code.

For me is a NO.

Reply:
@Mte90 Thank you. Could you share a code sample of how to do this and we can maybe well advise https://petitions.classicpress.net/posts/100/change-posts-to-articles-or-make-them-editable-so-they-can-be-changed-into-anything

Reply:
I saw now the petition, we are talking not changing the plural for new post type but for the default one.
It is possible with https://wpbeaches.com/change-the-wordpress-post-type-name-to-something-else/

Reply:
This is clear enough. We can make a snippet that we can share. Unless @nylen wants to pursue this further, we can close this. I will make a snippet or a side plugin for those non-techies to just drop into their CP.

Reply:
>The reason why I would go for a filter, is that when one adds it, they are doing it in their languages. I also presumed that the initial words stay as Post and Posts that can be translated.

This code would have to be written to filter each whole string separately, like this:

```
apply_filters( 'post_name_singular', __( 'Post' ) )
```

Or, build the array of all the post type labels, and then filter all the labels at once.

At that point there is not much benefit to the filter, since you can just grab the post type object after it is registered and manipulate it there, as in @Mte90's link.

The rule with `__` and similar functions is that they can only contain a literal, hard-coded string, not the result of any calculation.  So, any kind of attempt to automate this is going to run into problems with translations.

Two things that should be improved about that code snippet:

- Add `__( 'string', 'plugin-slug' )` calls for translation.
- Change the filter priority to `0` so these modifications run immediately after other hooks which run on `init` ([code reference](https://github.com/ClassicPress/ClassicPress/blob/b86cee8ce4b73d5afcd508d3ee21933ba6309992/src/wp-includes/default-filters.php#L436)).